python/pyarrow/_compute.pyx

```diff
@@ -2665,11 +2665,19 @@ cdef get_register_tabular_function():
     return reg
 
 
+cdef get_register_aggregate_function():
+    cdef RegisterUdf reg = RegisterUdf.__new__(RegisterUdf)
+    reg.register_func = RegisterAggregateFunction
+    return reg
+
+
 def register_scalar_function(func, function_name, function_doc, in_types, out_type,
                              func_registry=None):
     """
     Register a user-defined scalar function.
 
+    This API is EXPERIMENTAL.
```

Comment:
@westonpace Added this because I think this is fairly new/evolving API. Does marking at Experimental look ok to you?

Reply:
That's fine.  UDFs are also marked experimental in the [pyarrow docs](https://github.com/apache/arrow/blob/407423394cb988669a0b5716899810a924c0aa4c/docs/source/python/compute.rst?plain=1#L390).